### PR TITLE
PJones | fixes and cleanup

### PR DESCRIPTION
**Pull request by forksofpower**
- removed boilerplate firebase files
- fixed json indentation in prettier
- consolidated .gitignore's to a single file in root
- fixed a race condition during build due to the yarn `foreach` plugin running the build in parallel without waiting for shared libs and dependencies to finish building
- removed lerna and other unused deps

### CODE REVIEW

File: extensions/purchase-label/firebase.json [hunk 1]
```diff
@@ -1,17 +0,0 @@
...
-            "port": 5001
-        },
-        "firestore": {
-            "port": 8080
-        },
-        "ui": {
-            "enabled": true
-        }
-    }
-}
```
Comment (andyhite) [inline extensions/purchase-label/firebase.json:17]: Why are we removing the firebase.json? Are you sure the emulator still works without this? I was unable to get it to work without this previously.

File: extensions/rates/functions/src/config.ts [hunk 2]
```diff
@@ -1,9 +1,12 @@
+const splitAndTrimString = (value: string): string[] =>
+  value.split(',').map((id) => id.trim());
+
 export default {
   location: process.env.LOCATION!,
   databaseInstance: process.env.SELECTED_DATABASE_INSTANCE!,
   shipEngineApiKey: process.env.SHIPENGINE_API_KEY!,
   collectionPath: process.env.COLLECTION_PATH!,
-  carrierIds: process.env.CARRIER_IDS!.split(',').map((id) => id.trim()),
+  carrierIds: splitAndTrimString(process.env.CARRIER_IDS!),
```
Comment (andyhite) [inline extensions/rates/functions/src/config.ts:9]: TBH I preferred the way this was previously. Unnecessary abstraction until we actually have a need to split and trim strings multiple places in this file

File: extensions/purchase-label/firebase.json (hunk 1 above)
Comment (forksofpower) [reply extensions/purchase-label/firebase.json:17]: They're not needed and I found some examples of location `.env` and firebase config files near shared test setup files.

File: extensions/rates/functions/src/config.ts (hunk 2 above)
Comment (forksofpower) [reply extensions/rates/functions/src/config.ts:9]: yeah I saw that coming. It made me feel good. 👍

Comment (forksofpower) [reply extensions/rates/functions/src/config.ts]: updated!

**Pull request merged**